realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -213,33 +219,53 @@ public E get(int location) {
     }
 
     /**
-     * This method is not supported.
-     *
-     * @throws NoSuchMethodError always.
+     * {@inheritDoc}
      */
     @Override
-    public int indexOf(Object o) {
-        throw new NoSuchMethodError("indexOf is not supported on RealmResults");
+    public E first() {
+        if (size() > 0) {
+            return get(0);
+        } else {
+            throw new IndexOutOfBoundsException("No results was found.");
+        }
     }
 
     /**
-     * Gets the first object from the list.
-     *
-     * @return the first object.
-     * @throws ArrayIndexOutOfBoundsException if RealmResults is empty.
+     * {@inheritDoc}
      */
-    public E first() {
-        return get(0);
+    @Override
+    public E last() {
+        int size = size();
+        if (size > 0) {
+            return get(size - 1);
+        } else {
+            throw new IndexOutOfBoundsException("No results was found.");
+        }
     }
 
     /**
-     * Gets the last object from the list.
-     *
-     * @return the last object.
-     * @throws ArrayIndexOutOfBoundsException if RealmResults is empty.
+     * {@inheritDoc}
      */
-    public E last() {
-        return get(size()-1);
+    @Override
+    public void deleteFromRealm(int location) {
+        realm.checkIfValid();
```

Comment:
This is actually being done by TableView/Table through `isImmutable`. I modified the exception message to be more user friendly.